src/App.jsx

```diff
@@ -14,33 +17,93 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const FIELD_ALPHA = 'alphabet';
+const FIELD_LENGTH = 'length';
+const FIELD_REVERSE = 'reverse';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, sortField, reverseField) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    preparedGoods.sort((a, b) => {
+      switch (sortField) {
+        case FIELD_ALPHA:
+          return a.localeCompare(b);
+        case FIELD_LENGTH:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reverseField) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState('');
+  const list = getPreparedGoods(goodsFromServer, sortField, reverseField);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(FIELD_ALPHA)}
+          type="button"
+          className={cn('button', {
+            'is-info': true,
+            'is-light': sortField !== FIELD_ALPHA,
+          })}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          onClick={() => setSortField(FIELD_LENGTH)}
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== FIELD_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => {
+            return reverseField
+              ? setReverseField('')
+              : setReverseField(FIELD_REVERSE);
+          }}
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !reverseField,
+          })}
+        >
+          Reverse
+        </button>
+
+        {(reverseField || sortField) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setReverseField('');
+              setSortField('');
+            }}
```

Comment:
create an own function for this onClick handler and move these logic to a function